server/lib/report_server_web/live/report_run_live/show.ex

```diff
@@ -140,4 +172,64 @@ defmodule ReportServerWeb.ReportRunLive.Show do
     {:error, "Unknown file type or malformed data: #{filetype}"}
   end
 
+  defp check_query_state(report_run = %ReportRun{athena_query_id: athena_query_id}, live_view_pid) do
+    if poll_query_state?(report_run) do
+      with {:ok, athena_query_state, athena_result_url} <- AthenaDB.get_query_info(athena_query_id),
+           {:ok, report_run} <- Reports.update_report_run(report_run, %{athena_query_state: athena_query_state, athena_result_url: athena_result_url}) do
+        maybe_poll_query_state(report_run, live_view_pid)
+        report_run
+      else
+        _ ->
+          report_run
+      end
+    else
+      report_run
+    end
+  end
+
+  defp maybe_poll_query_state(report_run = %ReportRun{}, live_view_pid) do
+    if poll_query_state?(report_run) do
+      Process.send_after(live_view_pid, :poll_query_state, 1000)
```

Comment:
Is polling every second a lot?  Should we do some sort of backoff for long queries?

Reply:
The 1 second pooling is what we are currently doing and it works fine.  The problem with a backoff is folks will be waiting for these to complete the if we backoff they will be waiting way more than they need to wait.